### ISSUE TITLE
[catboost4j-spark] - "Error while executing workers" while using Catboost for Spark on Databricks Cluster

### ISSUE THREAD

**Issue by nofarm3** [catboost/catboost#1687]
Problem: Error while executing workers while using Catboost for Spark
catboost version: catboost-spark_3.0_2.12:0.25
Operating System: Databricks Cluster
CPU: 4 vcpu node (i3.xlarge servers)

I'm trying to run this code (the pyspark example from the README) in my notebook on Databricks cluster:

```
import catboost_spark
from pyspark.sql import SparkSession
from pyspark.ml.linalg import Vectors, VectorUDT
from pyspark.sql import Row
from pyspark.sql.types import *


srcDataSchema = [
    StructField("features", VectorUDT()),
    StructField("label", StringType())
]

trainData = [
    Row(Vectors.dense(0.1, 0.2, 0.11), "0"),
    Row(Vectors.dense(0.97, 0.82, 0.33), "1"),
    Row(Vectors.dense(0.13, 0.22, 0.23), "1"),
    Row(Vectors.dense(0.8, 0.62, 0.0), "0")
]
    
trainDf = spark.createDataFrame(spark.sparkContext.parallelize(trainData), StructType(srcDataSchema))
trainPool = catboost_spark.Pool(trainDf)
    
evalData = [
    Row(Vectors.dense(0.22, 0.33, 0.9), "1"),
    Row(Vectors.dense(0.11, 0.1, 0.21), "0"),
    Row(Vectors.dense(0.77, 0.0, 0.0), "1")
]
    
evalDf = spark.createDataFrame(spark.sparkContext.parallelize(evalData), StructType(srcDataSchema))
evalPool = catboost_spark.Pool(evalDf)

classifier = catboost_spark.CatBoostClassifier()

classifier.fit(trainPool, [evalPool])
```

I also changed these configurations to be the same (like mentioned in the README):
```
spark.task.cpus 4
spark.executor.cores 4
```

And I'm getting the following error:

```
---------------------------------------------------------------------------
Py4JJavaError                             Traceback (most recent call last)
<command-3098381892186308> in <module>
     32 classifier = catboost_spark.CatBoostClassifier()
     33 
---> 34 classifier.fit(trainPool, [evalPool])

/local_disk0/spark-9080518a-98d3-4c94-ae28-22ea83f8d868/userFiles-31f539c0-03af-431d-8d3f-b9608c016de6/addedFile6636251625245665267catboost_spark_3_0_2_12_0_25-126f6.jar/catboost_spark/core.py in fit(self, trainDataset, evalDatasets)
   4977                 evalDatasetsAsJavaObject[i] = _py2java(sc, evalDatasets[i])
   4978             self._transfer_params_to_java()
-> 4979             java_model = self._java_obj.fit(_py2java(sc, trainDataset), evalDatasetsAsJavaObject)
   4980             return CatBoostClassificationModel(java_model)
   4981 

/databricks/spark/python/lib/py4j-0.10.9-src.zip/py4j/java_gateway.py in __call__(self, *args)
   1303         answer = self.gateway_client.send_command(command)
   1304         return_value = get_return_value(
-> 1305             answer, self.gateway_client, self.target_id, self.name)
   1306 
   1307         for temp_arg in temp_args:

/databricks/spark/python/pyspark/sql/utils.py in deco(*a, **kw)
    125     def deco(*a, **kw):
    126         try:
--> 127             return f(*a, **kw)
    128         except py4j.protocol.Py4JJavaError as e:
    129             converted = convert_exception(e.java_exception)

/databricks/spark/python/lib/py4j-0.10.9-src.zip/py4j/protocol.py in get_return_value(answer, gateway_client, target_id, name)
    326                 raise Py4JJavaError(
    327                     "An error occurred while calling {0}{1}{2}.\n".
--> 328                     format(target_id, ".", name), value)
    329             else:
    330                 raise Py4JError(

Py4JJavaError: An error occurred while calling o320.fit.
: java.lang.RuntimeException: Error while executing workers
	at ai.catboost.spark.impl.Helpers$.checkOneFutureAndWaitForOther(Helpers.scala:30)
	at ai.catboost.spark.CatBoostPredictorTrait.fit(CatBoostPredictor.scala:206)
	at ai.catboost.spark.CatBoostPredictorTrait.fit$(CatBoostPredictor.scala:109)
	at ai.catboost.spark.CatBoostClassifier.fit(CatBoostClassifier.scala:362)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:380)
	at py4j.Gateway.invoke(Gateway.java:295)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.GatewayConnection.run(GatewayConnection.java:251)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.util.concurrent.ExecutionException: org.apache.spark.SparkException: Job aborted due to stage failure: Task 1 in stage 66.0 failed 4 times, most recent failure: Lost task 1.3 in stage 66.0 (TID 144, 10.231.252.40, executor 1): scala.MatchError: null
	at ai.catboost.spark.impl.CatBoostWorker.processPartition(Workers.scala:60)
	at ai.catboost.spark.impl.CatBoostWorkers.$anonfun$run$5(Workers.scala:241)
	at ai.catboost.spark.impl.CatBoostWorkers.$anonfun$run$5$adapted(Workers.scala:234)
	at org.apache.spark.rdd.RDD.$anonfun$foreachPartition$2(RDD.scala:1001)
	at org.apache.spark.rdd.RDD.$anonfun$foreachPartition$2$adapted(RDD.scala:1001)
	at org.apache.spark.SparkContext.$anonfun$runJob$5(SparkContext.scala:2379)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
	at org.apache.spark.scheduler.Task.doRunTask(Task.scala:144)
	at org.apache.spark.scheduler.Task.run(Task.scala:117)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$9(Executor.scala:640)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1581)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:643)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

Driver stacktrace:
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at ai.catboost.spark.impl.Helpers$.checkOneFutureAndWaitForOther(Helpers.scala:26)
	... 14 more
Caused by: org.apache.spark.SparkException: Job aborted due to stage failure: Task 1 in stage 66.0 failed 4 times, most recent failure: Lost task 1.3 in stage 66.0 (TID 144, 10.231.252.40, executor 1): scala.MatchError: null
	at ai.catboost.spark.impl.CatBoostWorker.processPartition(Workers.scala:60)
	at ai.catboost.spark.impl.CatBoostWorkers.$anonfun$run$5(Workers.scala:241)
	at ai.catboost.spark.impl.CatBoostWorkers.$anonfun$run$5$adapted(Workers.scala:234)
	at org.apache.spark.rdd.RDD.$anonfun$foreachPartition$2(RDD.scala:1001)
	at org.apache.spark.rdd.RDD.$anonfun$foreachPartition$2$adapted(RDD.scala:1001)
	at org.apache.spark.SparkContext.$anonfun$runJob$5(SparkContext.scala:2379)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
	at org.apache.spark.scheduler.Task.doRunTask(Task.scala:144)
	at org.apache.spark.scheduler.Task.run(Task.scala:117)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$9(Executor.scala:640)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1581)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:643)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

Driver stacktrace:
	at org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:2519)
	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:2466)
	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:2460)
	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
	at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:2460)
	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:1152)
	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:1152)
	at scala.Option.foreach(Option.scala:407)
	at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:1152)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:2721)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2668)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2656)
	at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)
	at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:938)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2339)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2360)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2379)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2404)
	at org.apache.spark.rdd.RDD.$anonfun$foreachPartition$1(RDD.scala:1001)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:165)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:125)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
	at org.apache.spark.rdd.RDD.withScope(RDD.scala:395)
	at org.apache.spark.rdd.RDD.foreachPartition(RDD.scala:999)
	at org.apache.spark.sql.Dataset.$anonfun$foreachPartition$1(Dataset.scala:2902)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at org.apache.spark.sql.Dataset.$anonfun$withNewRDDExecutionId$1(Dataset.scala:3698)
	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withCustomExecutionEnv$5(SQLExecution.scala:116)
	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:249)
	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withCustomExecutionEnv$1(SQLExecution.scala:101)
	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:845)
	at org.apache.spark.sql.execution.SQLExecution$.withCustomExecutionEnv(SQLExecution.scala:77)
	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:199)
	at org.apache.spark.sql.Dataset.withNewRDDExecutionId(Dataset.scala:3696)
	at org.apache.spark.sql.Dataset.foreachPartition(Dataset.scala:2902)
	at ai.catboost.spark.impl.CatBoostWorkers.run(Workers.scala:234)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	... 1 more
Caused by: scala.MatchError: null
	at ai.catboost.spark.impl.CatBoostWorker.processPartition(Workers.scala:60)
	at ai.catboost.spark.impl.CatBoostWorkers.$anonfun$run$5(Workers.scala:241)
	at ai.catboost.spark.impl.CatBoostWorkers.$anonfun$run$5$adapted(Workers.scala:234)
	at org.apache.spark.rdd.RDD.$anonfun$foreachPartition$2(RDD.scala:1001)
	at org.apache.spark.rdd.RDD.$anonfun$foreachPartition$2$adapted(RDD.scala:1001)
	at org.apache.spark.SparkContext.$anonfun$runJob$5(SparkContext.scala:2379)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
	at org.apache.spark.scheduler.Task.doRunTask(Task.scala:144)
	at org.apache.spark.scheduler.Task.run(Task.scala:117)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$9(Executor.scala:640)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1581)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:643)
	... 3 more
```

**Comment by btbbass** [comment]
Hi, 
I confirm the issue, I cannot run the example on a cluster in Scala with this:

Language: Scala Spark 2.4.4
catboost version: catboost-spark_2.4_2.11:0.25
Operating System: Toree Notebook on provate cloud

I am trying to run this code (from the tutorial)

```scala
import org.apache.spark.sql.{Row,SparkSession}
import org.apache.spark.ml.linalg._
import org.apache.spark.sql.types._

import ai.catboost.spark._

val srcDataSchema = Seq(
  StructField("features", SQLDataTypes.VectorType),
  StructField("label", StringType)
)

//training data containing features and label.
val trainData = Seq(
  Row(Vectors.dense(0.1, 0.2, 0.11), "0"),
  Row(Vectors.dense(0.97, 0.82, 0.33), "1"),
  Row(Vectors.dense(0.13, 0.22, 0.23), "1"),
  Row(Vectors.dense(0.8, 0.62, 0.0), "0")
)

val trainDf = spark.createDataFrame(spark.sparkContext.parallelize(trainData), StructType(srcDataSchema))
val trainPool = new Pool(trainDf)

//evaluation data containing features and label.
val evalData = Seq(
  Row(Vectors.dense(0.22, 0.33, 0.9), "1"),
  Row(Vectors.dense(0.11, 0.1, 0.21), "0"),
  Row(Vectors.dense(0.77, 0.0, 0.0), "1")
)

val evalDf = spark.createDataFrame(spark.sparkContext.parallelize(evalData), StructType(srcDataSchema))
val evalPool = new Pool(evalDf)

// train model
val classifier = new CatBoostClassifier 
val model = classifier.fit(trainPool, Array[Pool](evalPool))

```

And this is the error:


```
Name: java.lang.RuntimeException
Message: Error while executing workers
StackTrace:   at ai.catboost.spark.impl.Helpers$.checkOneFutureAndWaitForOther(Helpers.scala:30)
  at ai.catboost.spark.CatBoostPredictorTrait$class.fit(CatBoostPredictor.scala:206)
  at ai.catboost.spark.CatBoostClassifier.fit(CatBoostClassifier.scala:362)
  ... 48 elided
Caused by: java.util.concurrent.ExecutionException: org.apache.spark.SparkException: Job aborted due to stage failure: Task 1 in stage 1314.0 failed 4 times, most recent failure: Lost task 1.3 in stage 1314.0 (TID 925, 10.225.4.175, executor 2): scala.MatchError: null
	at ai.catboost.spark.impl.CatBoostWorker.processPartition(Workers.scala:60)
	at ai.catboost.spark.impl.CatBoostWorkers$$anonfun$run$3.apply(Workers.scala:235)
	at ai.catboost.spark.impl.CatBoostWorkers$$anonfun$run$3.apply(Workers.scala:234)
	at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1$$anonfun$apply$28.apply(RDD.scala:935)
	at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1$$anonfun$apply$28.apply(RDD.scala:935)
	at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:2101)
	at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:2101)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
	at org.apache.spark.scheduler.Task.run(Task.scala:123)
	at org.apache.spark.executor.Executor$TaskRunner$$anonfun$10.apply(Executor.scala:408)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1360)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:414)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Driver stacktrace:
  at java.util.concurrent.FutureTask.report(FutureTask.java:122)
  at java.util.concurrent.FutureTask.get(FutureTask.java:192)
  at ai.catboost.spark.impl.Helpers$.checkOneFutureAndWaitForOther(Helpers.scala:26)
  ... 50 more
Caused by: org.apache.spark.SparkException: Job aborted due to stage failure: Task 1 in stage 1314.0 failed 4 times, most recent failure: Lost task 1.3 in stage 1314.0 (TID 925, 10.225.4.175, executor 2): scala.MatchError: null
	at ai.catboost.spark.impl.CatBoostWorker.processPartition(Workers.scala:60)
	at ai.catboost.spark.impl.CatBoostWorkers$$anonfun$run$3.apply(Workers.scala:235)
	at ai.catboost.spark.impl.CatBoostWorkers$$anonfun$run$3.apply(Workers.scala:234)
	at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1$$anonfun$apply$28.apply(RDD.scala:935)
	at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1$$anonfun$apply$28.apply(RDD.scala:935)
	at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:2101)
	at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:2101)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
	at org.apache.spark.scheduler.Task.run(Task.scala:123)
	at org.apache.spark.executor.Executor$TaskRunner$$anonfun$10.apply(Executor.scala:408)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1360)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:414)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Driver stacktrace:
  at org.apache.spark.scheduler.DAGScheduler.org$apache$spark$scheduler$DAGScheduler$$failJobAndIndependentStages(DAGScheduler.scala:1889)
  at org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:1877)
  at org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:1876)
  at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
  at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
  at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:1876)
  at org.apache.spark.scheduler.DAGScheduler$$anonfun$handleTaskSetFailed$1.apply(DAGScheduler.scala:926)
  at org.apache.spark.scheduler.DAGScheduler$$anonfun$handleTaskSetFailed$1.apply(DAGScheduler.scala:926)
  at scala.Option.foreach(Option.scala:257)
  at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:926)
  at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:2110)
  at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2059)
  at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2048)
  at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)
  at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:737)
  at org.apache.spark.SparkContext.runJob(SparkContext.scala:2061)
  at org.apache.spark.SparkContext.runJob(SparkContext.scala:2082)
  at org.apache.spark.SparkContext.runJob(SparkContext.scala:2101)
  at org.apache.spark.SparkContext.runJob(SparkContext.scala:2126)
  at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1.apply(RDD.scala:935)
  at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1.apply(RDD.scala:933)
  at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
  at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
  at org.apache.spark.rdd.RDD.withScope(RDD.scala:363)
  at org.apache.spark.rdd.RDD.foreachPartition(RDD.scala:933)
  at org.apache.spark.sql.Dataset$$anonfun$foreachPartition$1.apply$mcV$sp(Dataset.scala:2741)
  at org.apache.spark.sql.Dataset$$anonfun$foreachPartition$1.apply(Dataset.scala:2741)
  at org.apache.spark.sql.Dataset$$anonfun$foreachPartition$1.apply(Dataset.scala:2741)
  at org.apache.spark.sql.Dataset$$anonfun$withNewRDDExecutionId$1.apply(Dataset.scala:3355)
  at org.apache.spark.sql.execution.SQLExecution$$anonfun$withNewExecutionId$1.apply(SQLExecution.scala:78)
  at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:125)
  at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:73)
  at org.apache.spark.sql.Dataset.withNewRDDExecutionId(Dataset.scala:3351)
  at org.apache.spark.sql.Dataset.foreachPartition(Dataset.scala:2740)
  at ai.catboost.spark.impl.CatBoostWorkers.run(Workers.scala:233)
  at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
  at java.util.concurrent.FutureTask.run(FutureTask.java:266)
  at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
  at java.util.concurrent.FutureTask.run(FutureTask.java:266)
  ... 3 more
Caused by: scala.MatchError: null
  at ai.catboost.spark.impl.CatBoostWorker.processPartition(Workers.scala:60)
  at ai.catboost.spark.impl.CatBoostWorkers$$anonfun$run$3.apply(Workers.scala:235)
  at ai.catboost.spark.impl.CatBoostWorkers$$anonfun$run$3.apply(Workers.scala:234)
  at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1$$anonfun$apply$28.apply(RDD.scala:935)
  at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1$$anonfun$apply$28.apply(RDD.scala:935)
  at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:2101)
  at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:2101)
  at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
  at org.apache.spark.scheduler.Task.run(Task.scala:123)
  at org.apache.spark.executor.Executor$TaskRunner$$anonfun$10.apply(Executor.scala:408)
  at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1360)
  at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:414)
```

**Comment by andrey-khropov** [comment]
> Language: Scala Spark 2.4.4
> catboost version: catboost-spark_2.4_2.11:0.25

I think the issue here is that Scala Spark 2.4.4 uses Scala 2.12, so you should be using `catboost-spark_2.4_2.12` package version.

The original poster's issue is likely unrelated.

**Comment by btbbass** [comment]
> > Language: Scala Spark 2.4.4
> > catboost version: catboost-spark_2.4_2.11:0.25
> 
> I think the issue here is that Scala Spark 2.4.4 uses Scala 2.12, so you should be using `catboost-spark_2.4_2.12` package version.
> 
> The original poster's issue is likely unrelated.

Well, my version of spark 2.4.4 is using scala 2.11 (it was a valid option), and other libraries works fine with the 2.11 flavour  on the same installation

**Comment by RmtcAF** [comment]
Hi, Just want to confirm that I have the same issue.

Environment: AWS EMR-5.30.1.
This EMR version has Spark 2.4.5, but it also uses 2.11, which is confirmed in spark shell.
Catboost version: catboost-spark_2.4_2.11:0.26
Node: r4.2xlarge

I tested the binary classifier code from readme:
https://github.com/catboost/catboost/tree/v0.26/catboost/spark/catboost4j-spark

In both Pyspark and Scala, I got the same error shared earlier.

```
Caused by: scala.MatchError: null
        at ai.catboost.spark.impl.CatBoostWorker.processPartition(Workers.scala:60)
 
```

**Comment by mrushad** [comment]
@nofarm3 were you able to resolve the issue?

**Comment by yzhang1-chwy** [comment]
I have the same issue on AWS emr-6.1.0 with spark 3.0.0 and scala 2.12

**Comment by praateekmahajan** [comment]
Same issue on Databricks Runtime 7.6 (Scala 2.12, Spark 3.0.1) and `ai.catboost:catboost-spark_3.0_2.12:0.26`.
![image](https://user-images.githubusercontent.com/7589415/126538230-26dbd672-fc18-41e4-b137-427091a513b5.png)

**Comment by nofarm3** [comment]
@mrushad 
hi, no..

**Comment by Devin-Taylor** [comment]
I am running into the same issue with Databricks 7.3 LTS ML, Spark 3.0.1, Scala 2.12, `ai.catboost:catboost-spark_3.0_2.12:0.26`. Has anyone had any success in finding a resolution/workaround?

**Comment by github-saad-elmahfoudi** [comment]
Same here: 
Environment: 
AWS emr-5.26.0
Spark 3.0.1, Scala 2.12
Catboost-spark 3.0: 0.26.1

Error after running the code from the tutorial:
```
scala.MatchError: null
	at ai.catboost.spark.impl.CatBoostWorker.processPartition(Workers.scala:60)
```

**Comment by andrey-khropov** [comment]
Fixed and deployed in release 1.0.3.
